apps/web/src/components/task/task-activities.tsx

```diff
@@ -38,22 +54,67 @@ function TaskActivities() {
               )}
             </div>
           </div>
-          <div className="flex-1">
+          <div className="flex-1 flex flex-col gap-2">
             <div className="flex items-center gap-2">
-              <span className="text-sm font-medium text-zinc-900 dark:text-zinc-100">
+              <span className="text-sm font-medium text-zinc-900 dark:text-zinc-400">
                 {activity.userEmail}
               </span>
               <span className="text-xs text-zinc-500 dark:text-zinc-400">
                 {formatDistanceToNow(activity.createdAt)} ago
               </span>
             </div>
-            <div className="mt-1.5">
-              <div className="text-sm text-zinc-600 dark:text-zinc-400">
+            <div className="flex flex-col gap-1">
+              <div className="text-sm text-zinc-600 dark:text-zinc-100">
                 {activity.type === "create" && activity.content}
                 {activity.type === "update" && activity.content}
                 {activity.type === "status" && activity.content}
-                {activity.type === "comment" && activity.content}
+                {activity.type === "comment" &&
+                  (editingCommentId === activity.id ? (
+                    <TaskComment
+                      initialComment={activity.content || ""}
+                      commentId={activity.id || null}
+                      onSubmit={() => setEditingCommentId(null)}
+                    />
+                  ) : (
+                    <span className="text-zinc-600 dark:text-zinc-100">
+                      {activity.content}
+                    </span>
+                  ))}
               </div>
+              {editingCommentId === null ? (
+                <div className="flex flex-row gap-3">
+                  <button
+                    type="button"
+                    className="text-xs underline text-zinc-500 dark:text-zinc-400"
+                    onClick={() => setEditingCommentId(activity.id ?? null)}
+                  >
+                    Edit
+                  </button>
+                  <button
+                    className="text-xs underline text-zinc-500 dark:text-zinc-400"
+                    tabIndex={0}
+                    type="button"
+                    onClick={() => handleDeleteComment(activity.id)}
+                    onKeyDown={(e) => {
+                      if (e.key === "Enter" || e.key === " ") {
+                        handleDeleteComment(activity.id);
+                      }
+                    }}
+                  >
+                    Delete
+                  </button>
+                </div>
+              ) : (
+                <div className="flex flex-row gap-3">
```

Comment:
I tried to reuse the TaskComment, but I'm not really a fan of the Cancel button's offset.
<img width="930" alt="image" src="https://github.com/user-attachments/assets/fd88b68c-84a0-46d7-9efe-b7b13c31f4ab" />

Reply:
Makes sense.